### ISSUE TITLE
NVFuser produces incorrect output for broadcasted dropout

### ISSUE THREAD

**Issue by ngimel** [pytorch/pytorch#82784]
```
import torch
import functorch
from functorch.compile import memory_efficient_fusion

def fn_drop(x, y):
    return torch.nn.functional.dropout(x) * y

x=torch.ones(4, device="cuda")
y=torch.ones(4,4,device="cuda")
opt_fn=memory_efficient_fusion(fn_drop)
for _ in range(100):
    out=opt_fn(x, y)
```

prints an output with randomly dropped out values instead of dropped out columns. Same for direct ts integration, don't know if it's the same for prim-based stack. 

cc @ezyang @gchanan @zou3519 @jjsjann123, @kevinstephano, @SherlockNoMad

**Comment by jjsjann123** [comment]
Oops, I'm guess there's something with scheduling. It also repros on a simple TS without functorch.

**Comment by jjsjann123** [comment]
linking issue: https://github.com/csarofeen/pytorch/pull/1888

**Comment by csarofeen** [comment]
It's codegen related, we need to break RNG offset from thread indexing, and use the position in the tensor we're filling. I think we should be able to get something in a few days. Targeting around Monday for a fix.

**Comment by csarofeen** [comment]
Fix is just about ready to go. @zasdfgbnm and @shmsong redid our RNG to make it both partially recomputable for all tensors (so we correctly recompute RNG values on broadcasted tensors) and completely independent of the parallelization scheme. Will get it pushed upstream ASAP.
